src/yargs-commands/schema/push.mjs

```diff
@@ -0,0 +1,122 @@
+import { container } from '../../cli.mjs'
+import { confirm } from "@inquirer/prompts";
+import { commonQueryOptions } from '../../lib/command-helpers.mjs';
+
+async function doPush(argv) {
+  const logger = (await container.resolve("logger"))
+  const fetch = container.resolve("fetch")
+
+  const gatherFSL = container.resolve("gatherFSL")
+  try {
+    const fsl = await gatherFSL(argv.dir)
+    if (argv.force) {
+      const params = new URLSearchParams()
+      if (argv.force)
+        params.set("force", "true")
+      if (argv.staged)
+        params.set("staged", "true")
+
+      const path = new URL(`/schema/1/update?${params}`, argv.url);
+      const res = await fetch(path.toString(), {
+        method: "POST",
+        headers: { AUTHORIZATION: `Bearer ${argv.secret}` },
+        body: fsl,
+        // https://github.com/nodejs/node/issues/46221
+        // https://github.com/microsoft/TypeScript-DOM-lib-generator/issues/1483
+        // @ts-expect-error-next-line
+        duplex: "half",
+      });
+
+      const json = await res.json();
+      if (json.error) {
+        logger.stderr(json.error?.message ?? json.error);
+      }
+    } else {
+      // Confirm diff, then push it. `force` is set on `validate` so we don't
+      // need to pass the last known schema version through.
+        const params = new URLSearchParams({ force: true })
+      if (argv.color)
+        params.set("color", "ansi")
+      const path = new URL(`/schema/1/validate?${params}`, argv.url);
+      const res = await fetch(path, {
+        method: "POST",
+        headers: { AUTHORIZATION: `Bearer ${argv.secret}` },
+        body: fsl,
+        // @ts-expect-error-next-line
+        duplex: "half",
+      });
+
+      const json = await res.json();
+      if (json.error) {
+        logger.stderr(json.error?.message ?? json.error);
```

Comment:
if this fails, the command should exit. does `logger.stderr` cause the command to exit immediately?

Reply:
it does not - the schema commands right now are very WIP.